.ci/Jenkinsfile

```diff
@@ -93,10 +104,55 @@ pipeline {
         }
       }
     }
+    stage('Release') {
+      options { skipDefaultCheckout() }
+      when {
+        beforeAgent true
+        tag pattern: 'v\\d+\\.\\d+.*', comparator: 'REGEXP'
+      }
+      stages {
+        stage('Dist') {
+          steps {
+            echo 'TBD'
+          }
+        }
+        stage('Release Notes') {
+          steps {
+            withGhEnv(forceInstallation: true, version: '2.4.0') {
+              dir("${BASE_DIR}"){
+                cmd(label: 'make release-notes', script: 'make -C .ci release-notes')
+              }
+            }
+          }
+        }
+      }
+      post {
```

Comment:
IIRC the post-stage is executed only if the stage is executed so because the release stage is only executed on tags the `whenTrue` conditionals can be removed